src/trex/dataset.py

```diff
@@ -62,42 +73,56 @@ def read_all(
         assert n_transcriptome == len(names)
 
         if n_transcriptome == 1:
-            reads = self.read_one(
-                transcriptome_inputs[0],
-                self.output_dir / "entries.bam",
-                cell_id_tag,
-                require_umis,
-            )
-            if n_amplicon == 1:
+            files = self.find_paths(transcriptome_inputs[0], cell_id_tag)
+            reads = []
+            for file in files:
                 reads.extend(
                     self.read_one(
-                        amplicon_inputs[0],
-                        self.output_dir / "amplicon_entries.bam",
+                        file,
+                        self.output_dir / "entries.bam",
                         cell_id_tag,
                         require_umis,
                     )
                 )
+            if n_amplicon == 1:
+                files = self.find_paths(amplicon_inputs[0], cell_id_tag)
+                for file in files:
+                    reads.extend(
+                        self.read_one(
+                            amplicon_inputs[0],
+                            self.output_dir / "amplicon_entries.bam",
+                            cell_id_tag,
+                            require_umis,
+                        )
+                    ) 
         else:
             datasets = []
             for *paths, name in zip_longest(
                 transcriptome_inputs, amplicon_inputs, names
             ):
                 assert name is not None
-                reads = self.read_one(
-                    paths[0],
-                    self.output_dir / (name + "_entries.bam"),
-                    cell_id_tag,
-                    require_umis,
-                )
-                if paths[1]:
-                    reads.extend(
+                files = self.find_paths(paths[0], cell_id_tag)
+                reads = []
+                for file in files:
+                    reads.exend(
                         self.read_one(
```

Comment:
This pattern of adding a loop around `self.read_one` is repeated. I think it would be nicer to add a `read_multiple()` function that does the looping and returns a single list with all reads.